src/merge.ts

```diff
@@ -235,18 +246,20 @@ export function mergeWithMapping(inFileContent: string, destFileContent: string,
             updateFirstAndLastChild(destUnit);
         } else {
             console.debug(`adding element with id "${unit.attr.id}"`);
-            if (options?.newTranslationTargetsBlank !== 'omit') {
-                const shouldBeBlank = (options?.newTranslationTargetsBlank ?? false) && !(options?.sourceLanguage ?? false);
-                const targetNode = new XmlDocument(`<target>${shouldBeBlank ? '' : unitSourceText}</target>`);
-                if (xliffVersion === '2.0') {
-                    const segmentSource = unit.childNamed('segment')!;
-                    segmentSource.children.push(targetNode);
-                } else {
-                    const sourceIndex = unit.children.indexOf(unitSource);
-                    unit.children.splice(sourceIndex + 1, 0, targetNode);
+            if (!getTargetElement(unit)) {
```

Comment:
why this new if?

Reply:
This is to avoid adding a second target node to the dest unit if it already has one. Try to comment it out, the test "should take only one target" will fail.